File: docs/src/testers/functions/management/filter_active_runs.ts
// @sniptest filename=filter_active_runs.ts
// @sniptest show=8-19
import assert from 'node:assert/strict';
import { NotteClient as FixtureClient } from 'notte-sdk';

const functionId = process.env.NOTTE_FUNCTION_ID!;
const fixture = new FixtureClient().NotteFunction({ function_id: functionId });
const completed = await fixture.run({ url: 'https://example.com' }, { stream: false });

import { NotteClient } from 'notte-sdk';

const client = new NotteClient();

// Get only active runs
const activeRuns = await client.NotteFunction({ function_id: functionId }).runs({ only_active: true });

console.log(`Active runs: ${activeRuns.items.length}`);

for (const run of activeRuns.items) {
  console.log(`Run ${run.function_run_id} - ${run.status}`);
}

const history = await fixture.runs({ only_active: false });
assert.ok(history.items.some(run => run.function_run_id === completed.function_run_id));
const results = activeRuns.items.map(run => run.status);
assert.deepEqual(results, []);
export { results };

File: docs/src/testers/functions/management/filter_active_runs.py
# @sniptest filename=filter_active_runs.py
# @sniptest show=9-19
# ruff: noqa: E402 -- fixture setup precedes the unchanged displayed import
import os

from notte_sdk import NotteClient as FixtureClient

function_id = os.environ["NOTTE_FUNCTION_ID"]
completed = FixtureClient().Function(function_id).run(url="https://example.com", stream=False)

from notte_sdk import NotteClient

client = NotteClient()

# Get only active runs
active_runs = client.functions.list_runs(function_id, only_active=True)

print(f"Active runs: {len(active_runs.items)}")

for run in active_runs.items:
    print(f"Run {run.workflow_run_id} - {run.status}")

history = client.functions.list_runs(function_id, only_active=False)
assert any(run.function_run_id == completed.function_run_id for run in history.items)
assert not active_runs.items
results = [run.status for run in active_runs.items]
